### ISSUE TITLE
Add downgrade cancellation e2e tests

### ISSUE THREAD

**Issue by henrybear327** [etcd-io/etcd#17976]
### What would you like to be added?

Add e2e tests for downgrade cancellation.

### Why is this needed?

Improves test coverage for downgrade procedure which is critical for implementing downgrade support.

**Comment by ahrtr** [comment]
@henrybear327 are you still working on this?

**Comment by henrybear327** [comment]
> @henrybear327 are you still working on this?

Yes! I have a branch that I never fully finish, sorry. I will polish it and submit it for review ASAP.

**Comment by ahrtr** [comment]
Discussed with @serathius, the intention is to ensure the cluster can be reverted to the original state in case any error happens during the downgrade.

**Comment by henrybear327** [comment]
> [@henrybear327](https://github.com/henrybear327) any update on this?

Had a call with @siyuanfoundation today and discussed. PR will be up this weekend.

**Comment by henrybear327** [comment]
Hey @ahrtr @siyuanfoundation, 

First part of the e2e test is delivered here https://github.com/etcd-io/etcd/pull/19244. 

The partial downgraded cluster cancellation PR will be delivered by the end of today, as I have a bug with it. Sorry that I probably should have broken it down when I was debugging so it won't look like there is no progress made so far...

**Comment by henrybear327** [comment]
The partial downgraded cluster cancellation PR is delivered here https://github.com/etcd-io/etcd/pull/19252

**Comment by henrybear327** [comment]
@siyuanfoundation @ahrtr I think we can close this issue after we either close or merge https://github.com/etcd-io/etcd/pull/19399 :)

Thank you!
